Mapping/Loader/YamlLoader.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Algolia\AlgoliaSearchBundle\Mapping\Loader;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Description;
+use Algolia\AlgoliaSearchBundle\Mapping\Index;
+use Algolia\AlgoliaSearchBundle\Mapping\IndexIf;
+use Algolia\AlgoliaSearchBundle\Mapping\Method;
+use Algolia\AlgoliaSearchBundle\Mapping\Property;
+use Symfony\Component\Yaml\Yaml;
+
+class YamlLoader extends AbstractFileLoader
+{
+    protected function loadMetadataFromFile(\ReflectionClass $class, $file)
+    {
+        $config = Yaml::parse(file_get_contents($file));
+
+        if (!isset($config[$name = $class->name])) {
+            throw new \RuntimeException(sprintf('Expected metadata for class %s to be defined in %s.', $class->name, $file));
+        }
+
+        $config = $config[$name];
+        $className = $this->removeProxy($name);
+        $description = new Description($class);
+
+        $this->extractClassMetadatas($config, $className, $description);
+
+        $this->extractPropertyMetadata($class, $className, $config, $description);
+
+        $this->extractMethodMetadata($class, $className, $config, $description);
+
+        return $description;
+    }
+
+    /**
+     * @param $config
+     * @param $className
+     * @param $description
+     */
+    protected function extractClassMetadatas($config, $className, Description $description)
```

Comment:
Naming: all other classes are called "extract...Metadata", this one has an extra "s" at the end.